### ISSUE TITLE
delta generation: panic on missing file

### ISSUE THREAD

**Issue by aaronovz1** [clearlinux/mixer-tools#578]
For whatever reason, when generating delta packs there are some files missing from the previous version full chroot, and this causes mixer to crash with a panic.

Should or can mixer not fail on this and just continue to generate deltas for other files that do exist?

```
| Found 1 previous versions
| Using 1 version threads and 8 delta threads in each
| panic: lstat /home/clr/mix/update/image/6/full/etc/gconf/gconf.xml.defaults/schemas/%gconf.xml: no such file or directory
| 
| goroutine 56 [running]:
| github.com/clearlinux/mixer-tools/swupd.renameDetection(0xc000158000, 0xc0001b5980, 0x2f, 0x2f, 0xc0019d6000, 0x1a7, 0x1a7, 0xc00051a040, 0x14, 0xc00051a100, ...)
| 	/builddir/build/BUILD/mixer-tools-5.6.0/.gopath/src/github.com/clearlinux/mixer-tools/swupd/rename.go:32 +0x6dd
| github.com/clearlinux/mixer-tools/swupd.linkDeltaPeersForPack(0xc000522000, 0xc00045c8c0, 0xc000158000, 0x26, 0xc00055a150)
| 	/builddir/build/BUILD/mixer-tools-5.6.0/.gopath/src/github.com/clearlinux/mixer-tools/swupd/manifest.go:577 +0xbac
| github.com/clearlinux/mixer-tools/swupd.findDeltas(0xc000522000, 0xc00045c8c0, 0xc000158000, 0xc0000001b6, 0xc0001b2030, 0x0, 0x0, 0x19)
| 	/builddir/build/BUILD/mixer-tools-5.6.0/.gopath/src/github.com/clearlinux/mixer-tools/swupd/delta.go:329 +0xaa
| github.com/clearlinux/mixer-tools/swupd.createDeltasFromManifests(0xc000522000, 0xc00045c8c0, 0xc000158000, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0)
| 	/builddir/build/BUILD/mixer-tools-5.6.0/.gopath/src/github.com/clearlinux/mixer-tools/swupd/delta.go:87 +0x1bc
| github.com/clearlinux/mixer-tools/swupd.CreateAllDeltas(0xc000476800, 0x18, 0x6, 0x7, 0x8, 0x0, 0x0)
| 	/builddir/build/BUILD/mixer-tools-5.6.0/.gopath/src/github.com/clearlinux/mixer-tools/swupd/packs.go:108 +0x3ac
| github.com/clearlinux/mixer-tools/builder.(*Builder).BuildDeltaPacksPreviousVersions.func1(0xc00051dd70, 0xc00007ed80, 0xc000476800, 0x18, 0xc00045c640, 0xc000126200, 0xc000470cc0)
| 	/builddir/build/BUILD/mixer-tools-5.6.0/.gopath/src/github.com/clearlinux/mixer-tools/builder/builder.go:581 +0xcf
| created by github.com/clearlinux/mixer-tools/builder.(*Builder).BuildDeltaPacksPreviousVersions
| 	/builddir/build/BUILD/mixer-tools-5.6.0/.gopath/src/github.com/clearlinux/mixer-tools/builder/builder.go:578 +0x841
| WARNING: exit code 2 from a shell command.
```

**Comment by rchiossi** [comment]
I can tell that it should most definitely not use `panic()` but rather use a proper error message.

Now, about continuing the build: Since delta packs are optional, it would be ok to ship generated packs even if  some are missing. What is not ok is to finish without informing the user that something went wrong. Just printing a warning wouldn't be enough, since it makes automation dependent on stdout.
If we want to implement this feature, the way to go is to have it under a flag (e.g. `--force` or `--ignore-missing`).

**Comment by aaronovz1** [comment]
That sounds good to me!

**Comment by aaronovz1** [comment]
@rchiossi  I've stumbled upon another problem as part of this same error. It looks like deltas are trying to be generated for files that no longer exist in the current version. ie if a file existed in version 1 but was deleted in version 2, then the panic error occurs.

**Comment by aaronovz1** [comment]
Any further progress on this?

**Comment by ashleshaAtrey** [comment]
@aaronovz1 are you still facing this issue? Are you seeing a warning or a panic?

**Comment by aaronovz1** [comment]
@ashleshaAtrey still in the process of updating mixer to include the fix for panic, however from what I can tell the change replaces panic with an error. Does returning an error skip generation for that one file and continue delta generation for the rest? @rchiossi had mentioned adding options such as `--ignore-missing`  etc

**Comment by ashleshaAtrey** [comment]
@aaronovz1 yes, even if it errors out, it will continue generating other deltas as delta creation is optional. You don't need any flag as this is the default behavior. Errors are printed for logging purpose. 
Let me know if you are still facing any problems.

**Comment by ashleshaAtrey** [comment]
Issue fixed in PR #599. Closing issue.
